src/test/java/org/elasticsearch/cluster/block/ClusterBlockTests.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cluster.block;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.io.stream.BytesStreamInput;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.test.ElasticsearchTestCase;
+import org.junit.Test;
+
+import java.util.EnumSet;
+
+import static org.elasticsearch.cluster.block.ClusterBlockLevel.*;
+import static org.hamcrest.CoreMatchers.equalTo;
+
+public class ClusterBlockTests extends ElasticsearchTestCase {
+
+    private static ClusterBlockLevel[] ALL_CLUSTER_BLOCK_LEVELS = ClusterBlockLevel.ALL.toArray(new ClusterBlockLevel[ALL.size()]);
+
+    @Test
+    public void testSerialization() throws Exception {
+        int iterations = randomIntBetween(5, 20);
+        for (int i = 0; i < iterations; i++) {
+            // Get a random version
+            Version fromVersion = randomVersion();
+            Version toVersion = randomVersion();
+
+            ClusterBlockLevel[] blockLevels = ALL_CLUSTER_BLOCK_LEVELS;
+            if (fromVersion.before(Version.V_1_6_0)) {
+                blockLevels = new ClusterBlockLevel[]{READ, WRITE, METADATA};
```

Comment:
I find this block a bit confusing, as well as `fromVersion` because it doesn't get used to serialize. I do see why you are doing this, but maybe it is not needed, we can just randomize the blocks (deprecated one will get mixed up with new ones, no big deal). What do you think? And rename `toVersion` to just `version`?